### PR TITLE
refactor: re-organize SDK and improve multi-API version support

### PR DESCRIPTION
**Pull request by lander2k2**

### CODE REVIEW

**Comment by lander2k2** [comment]
Add developer docs to explain object version vs library versioning.

File: docs/design/threeport-extensions.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Requirements:
+
+* The Threeport SDK must support extensions that are developed in their own repo,
+  outside the threeport/threeport repo and added to running Threeport control
+  plane.
+* Any number of Threeport extensions must be able to be added that are unknown
+  at the time the core Threeport control plane is deployed.
+
+Implications:
+* We will need to provide a namespacing mechanixm in the Threeport API that
```
Comment (randalljohnson) [inline docs/design/threeport-extensions.md:17]:
```suggestion
* We will need to provide a namespacing mechanism in the Threeport API that
```

File: docs/img/ThreeportLibVersioningEnvCompatibility.png
Comment (randalljohnson) [inline docs/img/ThreeportLibVersioningEnvCompatibility.png:1]:
```
mv Procuct Product
```

File: docs/versions.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+* SDK config: The SDK config is used by Threeport devs and Threeport extension
+  developers to define the objects in the system.  The object version hierachy
+  is as follows in the SDK config:
+  ```
+  Object Group
+  |-- Object
+      |-- Versions
+  ```
+  See the `sdk-config.yaml` for examples.
+* Codebase: In `pkg/api` of the Threeport codebase and its extentions, the object
```
Comment (randalljohnson) [inline docs/versions.md:87]:
```suggestion
* Codebase: In `pkg/api` of the Threeport codebase and its extensions, the object
```

File: docs/versions.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+If you have ever upgraded the versions of your Go library dependencies and had
+that break the code you've written that uses the library, you'll appreciate why
+we did this.
+
+We run into this pretty often.  It usually goes as follows:
+
+1. Upgrade depeendencies with `go get -u ./...`.
+1. New library versions are downloaded and recored in `go.mod`.
+1. Re-build project components and find compile errors.
+1. Find library function signatures that have chaned and update library usage in
```
Comment (randalljohnson) [inline docs/versions.md:122]:
```suggestion
1. Find library function signatures that have changed and update library usage in
```

File: docs/versions.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+If there is a feature or functionality they wish to use in `v1` they can make
+that upgrade with the import path in their code - but the version of Threeport
+recorded in `go.mod` does not need to change.
+
+### Get to the Point!
+
+We want to make upgrade paths as smooth as possible.
+
+With this library versionoing, your extensions and integrations with Threeport
```
Comment (randalljohnson) [inline docs/versions.md:152]:
```suggestion
With this library versioning, your extensions and integrations with Threeport
```

File: internal/control-plane/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+itself.
+
+## Genesis Control Plane
+
+A "genesis" control plane is a Threeport control plane that is provisioned with
+`tptctl up`.
+
+## Parent & Child Control Planes
+
+When a Threeprot control plane is used to spin up another, new Threeport control
```
Comment (randalljohnson) [inline internal/control-plane/README.md:14]:
```suggestion
When a Threeport control plane is used to spin up another, new Threeport control
```

**Review by randalljohnson — APPROVED** [review]

**Review by HarisChoudhary — APPROVED** [review]

**Pull request merged**